Packages/Tracking/OpenXR/Runtime/Scripts/XRLeapProviderManager.cs

```diff
@@ -0,0 +1,58 @@
+using Leap;
+using Leap.Unity;
+using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.XR.Management;
+using UnityEngine.XR.OpenXR;
+using Ultraleap.Tracking.OpenXR;
+
+
+public class XRLeapProviderManager : PostProcessProvider
+{
+    [SerializeField] private OpenXRLeapProvider openXRLeapProvider;
+    [SerializeField] private LeapXRServiceProvider leapXRServiceProvider;
+
+    public Action<LeapProvider> OnProviderSet;
+
+    private LeapProvider _leapProvider = null;
+    public LeapProvider LeapProvider
+    {
+        get
+        {
+            return (_leapProvider == null) ? leapXRServiceProvider : _leapProvider;
+        }
+    }
+
+    [Tooltip("Forces the use of the non-OpenXR provider, using LeapC hand tracking")]
+    public bool forceLeapService;
+
+    private IEnumerator Start()
+    {
+        while (XRGeneralSettings.Instance == null) yield return new WaitForEndOfFrame();
+
+        if (XRGeneralSettings.Instance.Manager.activeLoader.name == "Open XR Loader" &&
+            OpenXRSettings.Instance.GetFeature<HandTrackingFeature>() != null &&
+            OpenXRSettings.Instance.GetFeature<HandTrackingFeature>().enabled &&
+            !forceLeapService)
+        {
+            Debug.Log("Using OpenXR for Hand Tracking");
+            _leapProvider = openXRLeapProvider;
+            Destroy(leapXRServiceProvider.gameObject);
+        }
+        else
+        {
+            Debug.Log("Using LeapService for Hand Tracking");
+            _leapProvider = leapXRServiceProvider;
+            Destroy(openXRLeapProvider.gameObject);
```

Comment:
Will destroying the provider that is not needed sort the issue of other components that look for the provider or could they do this while there are two? Would we expect a client to access the post process provider to use this?

Reply:
They _could_ still get access and then it be deleted, as we need to wait for OpenXR to spin up, we can't push this class back in the order of execution to avoid this. We should advise devs to only reference the XRLeapProviderManager directly if they are using it to avoid finding invalid/unused Providers

Reply:
Ah yes, we'll need documentation for this. Good thing to put in that.